src/Cake.NuGet/Install/NuGetSourceRepositoryProvider.cs

```diff
@@ -34,10 +49,13 @@ public NuGetSourceRepositoryProvider(ISettings settings, ICakeConfiguration conf
                 CreateRepository(package.Address.AbsoluteUri);
             }
 
-            var nugetSource = config.GetValue(Constants.NuGet.Source);
-            if (!string.IsNullOrWhiteSpace(nugetSource))
+            var nugetSources = config.GetValue(Constants.NuGet.Source);
+            foreach (var nugetSource in nugetSources.Split(';'))
```

Comment:
Just curious, what will happen if `Constant.NuGet.Source` is not set in configuration? Will `ICakeConfiguration` return null?

Reply:
Seems like it will return `null`... We should guard for `null` here, otherwise we'll get a `NullReferenceException`
https://github.com/cake-build/cake/blob/develop/src/Cake.Core/Configuration/CakeConfiguration.cs#L35

Reply:
What if the nugetSource is actually null? Should we throw an exception for that, or is simply returning enough? My Hunch is, that returning should be ok, i am not sure though.